Automattic/vip-cli
# Subsite SQL Import into Launched Sites

## Description

* get isVIP from the public API so we can gate feature availability to staff during development
* displays the tables that will be imported as part of the SQL import process
* Prompts the user once, and adds friction by requiring the domain of the import

![image](https://user-images.githubusercontent.com/1752747/110743113-84d60d00-8205-11eb-82ec-3ee42e855645.png)

### Test Plan

#### Jest

`npm t` for automated tests.

#### Manual Test plan

1. Checkout this branch
1. Execute an import for an unlaunched non-multisite app `npm run build && ./dist/bin/vip-import-sql.js @{id}.develop ~/your-sql-dump.sql`
1. Note that the list of tables that will be imported are listed and that you must type the domain name of the app as confirmation.
2. Execute an import for a launched multisite app `npm run build && ./dist/bin/vip-import-sql.js @{id}.production ~/multisite-dump.sql`
3. Note that the list of tables that will be imported are listed and separated by blog_id and that confirmation to proceed requires you to type the domain name.
4. Logout of the CLI
5. Login as a non isVIP user
6. Retry the imported into a launched multisite and note that you cannot proceed 

Closes #691 
Closes #693 
Closes #695 
Closes #696 


---
_This PR was started by: [git wf pr](https://github.com/groupon/git-workflow/releases/tag/v1.3.2)_
